contributors/devel/sig-testing/monitoring.md

```diff
@@ -0,0 +1,264 @@
+```
+
+failed as well.
+
+If one or both of these jobs continue failing, or if they fail frequently
+enough, we should open an issue and investigate.
+
+3. **Since when has it been flaking**
+
+This information you can get from the header of the TestGrid page showing you all the
+tests.
+
+Going from top to bottom, you will see:
+* Time at which the job was started
+* Prow job run ID
+* Kubernetes commit that was tested
+* (Most often) test-infra commit
+
+You can also find this data in Triage see below.
+
+4. **Reason for failure**
+
+The aim for this issue is to begin investigating - you don't have to find the
+reason for failure right away (nor the solution). You can just log the error
+reported by the test when the job was run.
+
+One way of getting useful information is to click on the failed runs (the red
+rectangles) and you will be taken to a page that displays the log files from 
+the job run.
+
+If we do this for the above test failures in `node-kubelet-features-master`, we
+will see the following
+
+<img src="./testgrid-images/spyglass-summary.png" height="60%" width="100%">
+
+Right away it will show you what tests failed.
+
+Here we see that 2 tests failed (both related to the node problem detector) and
+the `e2e.go: Node Tests` stage was marked as failed (because the node problem
+detector tests failed).
+
+You will often see "stages" (steps in an e2e job) as mixed with the tests
+themselves. The stages tell you what was going on in the e2e job when an error
+occurred.
+
+If we click on the first test error, we will see logs that will (hopefully) help
+us figure out why the test failed.
+
+<img src="./testgrid-images/spyglass-result.png" height="60%" width="100%">
+
+Further down the page you will see all the logs for the entire test run.
+Please copy any information you think may be useful from here into the issue.
+
+You can reference a specific line in the logs by click on the line number and 
+then copying the url which will now include an anchor to the specific line.
+
+5. **Anything else we need to know**
+
+It is important to review the behavior of the flaking test across a range of jobs
+using https://go.k8s.io/triage
+
+We can use the Triage tool to see if a test we see failing in a given job has
+been failing in others and to understand how jobs are behaving.
+
+For example, we can see how the job we have been looking at has been behaving
+recently.
+
+There is one important detail we have to mention at this point, the job names
+you see on TestGrid are often aliases.
+
+For example, when we clicked on a test run for
+[node-kubelet-features-master](https://testgrid.k8s.io/sig-release-master-informing#node-kubelet-features-master),
+at the top left corner of Spyglass the page tells us the real job name,
+`ci-kubernetes-node-kubelet-features` (notice the "ci-kubernetes-" prefix).
```

Comment:
Ahhh, I see the comment I made about TestGrid tab names not being "real" job names has come up here. Not sure whether this should be moved further up to step 1?? but it requires the reader to know to click on the cells in TestGrid for Spyglass which we walk through in step 4?

Reply:
Echoing what I said above, this should practically be at the top, under a linkable heading like "Glossary" or "Terminology" or something

Reply:
Or reported as a UX defect.

Reply:
TestGrid has been improved on this point. (I think thanks to work done by @spiffxp ! Thank you Aaron) So I'll update the screen shot and adjust the copy accordingly.

Reply:
I've added an svg version of the screen shot. SVG files render fine on github markdown 

I created the svg file using Inkscape (quirky UX but powerful and worth learning effort) 

Inkscape diagrams have layers. Screenshot from TG is in it's own layer. The annotations I added pointing out TG UI elements (commit sources, Prow Job numbers, Prow Job config and desc) are in a separate layer. 

This means that we can reuse those annotations in other screen shots and also will make the screen shot a little more maintainable when we pick up TestGrid improvements in the future.